src/promptflow-core/promptflow/executor/_service/utils/process_utils.py

```diff
@@ -81,15 +82,19 @@ async def invoke_sync_function_in_process(
                 ProcessManager().remove_process(run_id)
 
 
-def _execute_target_function(
+def _set_environment_variables_then_execute_target_function(
     target_function: Callable,
     args: tuple,
     kwargs: dict,
     return_dict: dict,
     error_dict: dict,
     context_dict: dict,
+    environment_variables: dict,
 ):
     block_terminate_signal_to_parent()
+    if environment_variables:
```

Comment:
There is a util function `set_environment_variables` in `src\promptflow-core\promptflow\executor\_service\utils\service_utils.py`. You can use it.